File: LeeJaeWoo/code/src/hw2/movie/policy/AmountDiscountPolicy.java
package hw2.movie.policy;

import hw2.item.Money;
import hw2.movie.condition.discount.DiscountCondition;
import hw2.movie.screen.Screening;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class AmountDiscountPolicy implements DiscountPolicy{

    private final Money amount;

    private List<DiscountCondition> discountConditions = new ArrayList<>();

    public AmountDiscountPolicy(Money amount,DiscountCondition... discountConditions) {
        this.amount = amount;
        this.discountConditions = Arrays.asList(discountConditions);
    }

    public Money getCalculateDiscountAmount(Screening screening) {
        for (DiscountCondition discountCondition : discountConditions) {
            if (discountCondition.isSatisfied(screening)) {
                return calculateFee(screening);
            }
        }
        return Money.ZERO;
    }

    @Override
    public Money calculateFee(Screening screening) {
        return screening.getMovieFee().minusMoney(amount);
    }
}

File: ParkKwiNam/chapter2/ticketing/discount/policy/AmountDiscountPolicy.kt
package chapter2.ticketing.discount.policy

import chapter2.items.Money
import chapter2.ticketing.discount.condition.DiscountCondition

class AmountDiscountPolicy(private val amount: Money, discountCondition: List<DiscountCondition>) :
    ConditionDiscountPolicy(discountCondition) {
    override fun getDiscountMoney(fee: Money): Money {
        return Money(fee.value - amount.value)
    }
}